internal/watcher/watcher.go

```diff
@@ -18,9 +18,18 @@ var (
 	ErrBucketMismatch = errors.New("bucket mismatch")
 )
 
+type Cleanup func(ctx context.Context) error
+
+func noopCleanup(ctx context.Context) error {
+	return nil
+}
+
 type Watcher interface {
 	// Watch waits until a blob is dispatched.
-	Watch(ctx context.Context) (*BlobEvent, error)
+	// After the event is successfully
+	// processed by the receiver, the returned Cleanup function must be executed to
+	// mark the event as processed. Implementors must not return a nil function.
```

Comment:
The way you're wrapping the comment is weird, why is the line break right after successfully? I think it should be as follows:

```go
	// After the event is successfully processed by the receiver, the returned
	// Cleanup function must be executed to mark the event as processed.
	// Implementors must not return a nil function.
```

Reply:
make sense.